### ISSUE TITLE
Just one container started

### ISSUE THREAD

**Issue by skornehl** [jenkinsci/docker-plugin#103]
Hi,

I have an existing Jenkins (ver. 1.582) with the Docker Plugin (0.8) and configured 2 machines (Docker version:  1.1.2 - falling out of yum with REHL7)

I'm starting a job which starts 5 subjobs, all with the same tags which are only used on the docker container. Container Cap is set to "empty", no instance cap.

After start, there is exactly one container on the first machine started. If I shut down docker on the first machine, one container is starting on the second machine. So there is only one job running at a time. If I delete that tag, all 5 running simultaneously on the dump slaves, which are still configured.

Log saying not much:

> Oct 15, 2014 2:39:04 PM com.nirima.jenkins.plugins.docker.DockerCloud provision
INFO: Excess workload after pending Spot instances: 1
> Oct 15, 2014 2:39:04 PM com.nirima.jenkins.plugins.docker.DockerCloud countCurrentDockerSlaves
> INFO: Images found: [Image{repository=null, tag=null, id=13f9985bd5f80cfcbded74777cf2a709b4f09133e616ed408fb67232c3b0a9c1, created=1413275760, size=2158078667, virtualSize=2158078667}]
> Oct 15, 2014 2:39:04 PM com.nirima.jenkins.plugins.docker.DockerCloud addProvisionedSlave
INFO: Provisioning for AMI jenkins; Estimated number of total slaves: 0; Estimated number of slaves for ami jenkins: 0
Launching jenkins
> Oct 15, 2014 2:39:04 PM hudson.slaves.NodeProvisioner update
> INFO: Started provisioning Image of jenkins from cel03 with 1 executors. Remaining excess workload:0.0
> Oct 15, 2014 2:39:04 PM com.nirima.jenkins.plugins.docker.DockerCloud provision
> INFO: Excess workload after pending Spot instances: 0
Oct 15, 2014 2:39:04 PM com.nirima.jenkins.plugins.docker.DockerTemplateBase getLxcConf
WARNING: Specified option:  is not in the form X=Y, please correct.
Oct 15, 2014 2:39:05 PM com.nirima.jenkins.plugins.docker.DockerComputerLauncher getSSHLauncher
INFO: Creating slave SSH launcher for cel03:49158
[10/15/14 14:39:10] SSH Launch of 6912795d6794@cel03 on cel03 completed in 5,553 ms
Oct 15, 2014 2:39:15 PM hudson.slaves.NodeProvisioner update
INFO: Image of jenkins provisioningE successfully completed. We have now 12 computer(s)

Any Idea why not more containers are starting? Each Job takes about 20 minutes so it is not https://github.com/jenkinsci/docker-plugin/issues/50

Thanks for help!

**Comment by thomassuckow** [comment]
The provisioner is now extendable, may be worth considering making a provisioner for one-off clouds.
https://issues.jenkins-ci.org/browse/JENKINS-24752

**Comment by skahlhoefer** [comment]
Same problem here. Are there installations that work as expected? And if so, how are they configured in detail?
Maybe we have set up a wrong configuration but I don't see any other way running a job explicitely in an own Docker container than limiting it to the label.

**Comment by KostyaSha** [comment]
Every new commentor, please provide full description about core, plugins versions and etc. Because it difficult to understand what was changed and how reproduce

**Comment by skahlhoefer** [comment]
Sorry.
Jenkins Core 1.595
Docker Plugin 0.9.0-rc1
Docker 1.6.2
Set up Docker Cloud using Docker Plugin and one image with a specific Jenkins node label "docker-slave". Other (physical) nodes are configured as well but should not be used for this job. The Jenkins job itself is limited to nodes with the label "docker-slave" or alternatively parameterized using label "docker-slave". Both job configurations lead to the same behaviour, described in this thread.

**Comment by KostyaSha** [comment]
> Docker Plugin 0.9.0-rc1

Could update to latest?

**Comment by skahlhoefer** [comment]
Sorry about the delay. Meanwhile I could update the plugin to latest 0.9.3 and did some tests again.
Turns out, a second job (on a dedicated second container) is started parallel to the first but only after a delay of 2 minutes.
I set up a fresh test environment with Jenkins 1.609.1 and Docker plugin 0.9.3, nothing special configured and a project executing just a shell command to sleep 5 minutes.

So I admit, I'm not facing this bug but... Any ideas on how to fix / shrink this delay?
Thanks in advance.

**Comment by KostyaSha** [comment]
Could you try #234 ? I hope it should solve this bad provisioning issues.

**Comment by dreilly1982** [comment]
I have tried #234 and I am still seeing just one container provisioned at a time.

**Comment by dreilly1982** [comment]
I was mistaken.  Even in 0.9.3 I am able to provision multiple containers at a time.  It can take up to two minutes after the first if the queue is small, but it does in fact work as expected.

**Comment by KostyaSha** [comment]
Container running (i hope) was fixed in 0.10. Jenkins itself has 1 minute delay for checking provisioning plus provisioner itself may decide to check.

@thomassuckow if it pluggable and plugin needs create it's own implementation then please open new issue.
